tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml

```
@@ -0,0 +1,134 @@
+---
+## Task file for setup/teardown AWS resources for aws_ssm integration testing
+- name: 'aws_ssm connection plugin integration test resource creation'
+  collections:
+    - amazon.aws
+  module_defaults:
+    group/aws:
+      aws_access_key: '{{ aws_access_key }}'
+      aws_secret_key: '{{ aws_secret_key }}'
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region }}'
+  block:
+
+    - name: Ensure IAM instance role exists
+      iam_role:
+        name: "ansible-test-{{tiny_prefix}}-aws-ssm-role"
+        assume_role_policy_document: "{{ lookup('file','ec2-trust-policy.json') }}"
+        state: present
+        create_instance_profile: yes
+        managed_policy:
+        - AmazonSSMManagedInstanceCore
+      register: role_output
+
+    - name: Lookup AMI configuration
+      set_fact:
+        ami_configuration: '{{ ami_details[(target_os | default("fedora"))] }}'
+
+    - name: AMI Lookup
+      ec2_ami_info:
+        owners: '{{ ami_configuration.owner }}'
+        filters:
+          name: '{{ ami_configuration.name }}'
+      register: ec2_amis
+
+    - name: Set facts with latest AMIs
+      vars:
+        latest_ami: '{{ ec2_amis.images | sort(attribute="creation_date") | last }}'
+      set_fact:
+        latest_ami_id: '{{ latest_ami.image_id }}'
+
+    # (Local installation of the SSM **client** which is then used by the plugin)
+    - name: Install Session Manager Client for Debian/Ubuntu
+      include_tasks: debian.yml
+      when: ansible_distribution == "Ubuntu" or ansible_distribution == "Debian"
```

```suggestion
      when: ansible_distribution in ["Ubuntu",  "Debian"]
```

same construct like for rpm based as below. but that's not a blocker